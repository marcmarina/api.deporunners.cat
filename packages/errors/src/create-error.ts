export function createValidationError(
  errorMessage: string,
  path: string[],
): Record<string, unknown> {
  return path.reduceRight((prev, cur) => {
    return { [cur]: prev };
  }, errorMessage as unknown as Record<string, unknown>);
}
