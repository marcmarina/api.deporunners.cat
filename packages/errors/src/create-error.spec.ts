import { createValidationError } from './create-error';

describe('createValidationError', () => {
  it.each`
    message               | path
    ${'validation error'} | ${['a']}
    ${'validation error'} | ${['a', 'b']}
    ${'validation error'} | ${['a', 'b', 'c']}
  `('should return an error object', ({ message, path }) => {
    expect(createValidationError(message, path)).toMatchSnapshot();
  });
});
