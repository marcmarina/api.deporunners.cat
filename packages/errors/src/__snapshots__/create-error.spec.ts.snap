// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createValidationError should return an error object 1`] = `
{
  "a": "validation error",
}
`;

exports[`createValidationError should return an error object 2`] = `
{
  "a": {
    "b": "validation error",
  },
}
`;

exports[`createValidationError should return an error object 3`] = `
{
  "a": {
    "b": {
      "c": "validation error",
    },
  },
}
`;
